device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/FileUpload.java

```diff
@@ -27,50 +27,59 @@
         httpsTransportManager.open();
     }
 
-    FileUploadSasUriResponse getFileUploadSasUri(FileUploadSasUriRequest request) throws IOException
+    FileUploadSasUriResponse getFileUploadSasUri(FileUploadSasUriRequest request) throws IotHubClientException
     {
         IotHubTransportMessage message = new IotHubTransportMessage(request.toJson());
         message.setIotHubMethod(HttpsMethod.POST);
 
-        HttpsResponse responseMessage = httpsTransportManager.getFileUploadSasUri(message);
+        try
+        {
+            HttpsResponse responseMessage = httpsTransportManager.getFileUploadSasUri(message);
+            String responseMessagePayload = validateServiceStatusCode(responseMessage);
 
-        String responseMessagePayload = validateServiceStatusCode(responseMessage, "Failed to get the file upload SAS URI");
+            if (responseMessagePayload == null || responseMessagePayload.isEmpty())
+            {
+                throw new IotHubClientException(IotHubStatusCode.ERROR, "SAS URI response message had no payload");
+            }
 
-        if (responseMessagePayload == null || responseMessagePayload.isEmpty())
+            return new FileUploadSasUriResponse(responseMessagePayload);
+        }
+        catch (IOException e)
```

Comment:
If there is any other kind of exception that could be thrown that we don't anticipate, should we have a general exception type we catch and still translate into an `IotHubClientException`?

Reply:
For cases like "JVM out of memory" types of exceptions? That would be pretty atypical as far as I'm aware.